--- a/scripts/projectsmirrors.json
+++ b/scripts/projectsmirrors.json
@@ -3,73 +3,31 @@
 		"https://downloads.sourceforge.net"
 	],
 	"@DEBIAN": [
-		"https://mirrors.tencent.com/debian",
-		"https://mirrors.aliyun.com/debian",
-		"https://mirrors.tuna.tsinghua.edu.cn/debian",
-		"https://mirrors.ustc.edu.cn/debian",
 		"https://ftp.debian.org/debian",
-		"https://mirror.leaseweb.com/debian",
-		"https://mirror.netcologne.de/debian"
+		"https://mirror.leaseweb.com/debian"
 	],
 	"@APACHE": [
-		"https://mirrors.tencent.com/apache",
-		"https://mirrors.aliyun.com/apache",
-		"https://mirrors.tuna.tsinghua.edu.cn/apache",
-		"https://mirrors.ustc.edu.cn/apache",
 		"https://dlcdn.apache.org",
-		"https://mirror.aarnet.edu.au/pub/apache",
-		"https://mirror.csclub.uwaterloo.ca/apache",
-		"https://archive.apache.org/dist",
-		"https://mirror.cogentco.com/pub/apache",
-		"https://mirror.navercorp.com/apache",
-		"https://ftp.jaist.ac.jp/pub/apache",
-		"https://apache.cs.utah.edu/apache.org",
-		"http://apache.mirrors.ovh.net/ftp.apache.org/dist"
+		"https://archive.apache.org/dist"
 	],
         "@GITHUB": [
                 "https://raw.githubusercontent.com"
         ],
 	"@GNU": [
-		"https://mirrors.tencent.com/gnu",
-		"https://mirrors.aliyun.com/gnu",
-		"https://mirrors.tuna.tsinghua.edu.cn/gnu",
-		"https://mirrors.ustc.edu.cn/gnu",
 		"https://ftpmirror.gnu.org",
-		"https://mirror.csclub.uwaterloo.ca/gnu",
-		"https://mirror.netcologne.de/gnu",
-		"https://ftp.kddilabs.jp/GNU/gnu",
-		"https://www.nic.funet.fi/pub/gnu/gnu",
-		"https://mirror.navercorp.com/gnu",
-		"https://mirrors.rit.edu/gnu",
 		"https://ftp.gnu.org/gnu"
 	],
 	"@SAVANNAH": [
 		"https://download.savannah.nongnu.org/releases",
-		"https://mirror.netcologne.de/savannah",
-		"https://mirror.csclub.uwaterloo.ca/nongnu",
-		"https://ftp.acc.umu.se/mirror/gnu.org/savannah",
-		"https://nongnu.uib.no",
 		"https://cdimage.debian.org/mirror/gnu.org/savannah"
 	],
 	"@KERNEL": [
-		"https://mirror.iscas.ac.cn/kernel.org",
-		"https://mirrors.ustc.edu.cn/kernel.org",
-		"https://mirror.nju.edu.cn/kernel.org",
 		"https://cdn.kernel.org/pub",
 		"https://mirrors.mit.edu/kernel",
-		"http://ftp.nara.wide.ad.jp/pub/kernel.org",
-		"http://www.ring.gr.jp/archives/linux/kernel.org",
-		"https://ftp.riken.jp/Linux/kernel.org",
 		"https://www.mirrorservice.org/sites/ftp.kernel.org/pub"
 	],
 	"@GNOME": [
-		"https://mirrors.ustc.edu.cn/gnome/sources",
-		"https://mirror.nju.edu.cn/gnome",
-		"https://download.gnome.org/sources",
-		"https://mirror.csclub.uwaterloo.ca/gnome/sources",
-		"https://ftp.acc.umu.se/pub/GNOME/sources",
-		"http://ftp.cse.buffalo.edu/pub/Gnome/sources",
-		"http://ftp.nara.wide.ad.jp/pub/X11/GNOME/sources"
+		"https://download.gnome.org/sources"
 	],
 	"@OPENWRT": [
 		"https://sources.cdn.openwrt.org",
--- a/target/linux/ramips/dts/mt7628an_tplink_tl-wr840n-v6.2.dts	1970-01-01 07:00:00.000000000 +0700
+++ b/target/linux/ramips/dts/mt7628an_tplink_tl-wr840n-v6.2.dts
@@ -0,0 +1,148 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "mt7628an.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/leds/common.h>
+
+/ {
+	compatible = "tplink,tl-wr840n-v6.2", "mediatek,mt7628an-soc";
+	model = "TP-Link TL-WR840N v6.2";
+
+	aliases {
+		led-boot = &led_wlan;
+		led-failsafe = &led_wlan;
+		led-upgrade = &led_wlan;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,115200";
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&gpio 38 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		lan {
+			function = LED_FUNCTION_LAN;
+			color = <LED_COLOR_ID_GREEN>;
+			gpios = <&gpio 39 GPIO_ACTIVE_LOW>;
+		};
+
+		wan_green {
+			function = LED_FUNCTION_WAN;
+			color = <LED_COLOR_ID_GREEN>;
+			gpios = <&gpio 40 GPIO_ACTIVE_LOW>;
+		};
+
+		led_wlan: wlan {
+			function = LED_FUNCTION_WLAN;
+			color = <LED_COLOR_ID_GREEN>;
+			gpios = <&gpio 41 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0tpt";
+		};
+
+		wan_orange {
+			function = LED_FUNCTION_WAN;
+			color = <LED_COLOR_ID_ORANGE>;
+			gpios = <&gpio 42 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "boot";
+				reg = <0x0 0x10000>;
+				read-only;
+			};
+
+			partition@10000 {
+				compatible = "tplink,firmware";
+				label = "firmware";
+				reg = <0x10000 0xfe0000>;
+			};
+
+			partition@ff0000 {
+				label = "factory";
+				reg = <0xff0000 0x10000>;
+				read-only;
+
+				nvmem-layout {
+					compatible = "fixed-layout";
+					#address-cells = <1>;
+					#size-cells = <1>;
+
+					eeprom_factory_0: eeprom@0 {
+						reg = <0x0 0x400>;
+					};
+
+					macaddr_factory_f100: macaddr@f100 {
+						reg = <0xf100 0x6>;
+					};
+				};
+			};
+		};
+	};
+};
+
+&ehci {
+	status = "disabled";
+};
+
+&ohci {
+	status = "disabled";
+};
+
+&wmac {
+	status = "okay";
+
+	nvmem-cells = <&eeprom_factory_0>, <&macaddr_factory_f100>;
+	nvmem-cell-names = "eeprom", "mac-address";
+};
+
+&ethernet {
+	nvmem-cells = <&macaddr_factory_f100>;
+	nvmem-cell-names = "mac-address";
+};
+
+&esw {
+	mediatek,portmap = <0x3e>;
+};
+
+&state_default {
+	gpio {
+		groups = "p4led_an", "p3led_an", "p2led_an", "p1led_an", "p0led_an", "wdt";
+		function = "gpio";
+	};
+};
+
+&gpio {
+	led_wlan_enable {
+		gpio-hog;
+		gpios = <43 GPIO_ACTIVE_HIGH>;
+		output-high;
+	};
+};
--- a/target/linux/ramips/image/mt76x8.mk
+++ b/target/linux/ramips/image/mt76x8.mk
@@ -900,6 +900,20 @@
 endef
 TARGET_DEVICES += tplink_tl-wr840n-v5
 
+define Device/tplink_tl-wr840n-v6.2
+  $(Device/tplink-v2)
+  IMAGE_SIZE := 16384k
+  DEVICE_MODEL := TL-WR840N
+  DEVICE_VARIANT := v6.2
+  TPLINK_FLASHLAYOUT := 16Mmtk
+  TPLINK_HWID := 0x08400006
+  TPLINK_HWREVADD := 0x6
+  IMAGES := sysupgrade.bin tftp-recovery.bin
+  IMAGE/tftp-recovery.bin := pad-extra 64k | $$(IMAGE/factory.bin)
+  DEFAULT := n
+endef
+TARGET_DEVICES += tplink_tl-wr840n-v6.2
+
 define Device/tplink_tl-wr841n-v13
   $(Device/tplink-v2)
   IMAGE_SIZE := 7808k
--- a/target/linux/ramips/mt76x8/base-files/etc/board.d/01_leds
+++ b/target/linux/ramips/mt76x8/base-files/etc/board.d/01_leds
@@ -132,6 +132,7 @@
 	ucidef_set_led_switch "lan4" "lan4" "green:lan4" "switch0" "0x10"
 	ucidef_set_led_switch "wan" "wan" "green:wan" "switch0" "0x01"
 	;;
+tplink,tl-wr840n-v6.2|\
 tplink,tl-wr841n-v14)
 	ucidef_set_led_switch "lan" "lan" "green:lan" "switch0" "0x1e"
 	ucidef_set_led_switch "wan" "wan" "green:wan" "switch0" "0x01"
--- a/target/linux/ramips/mt76x8/base-files/etc/board.d/02_network
+++ b/target/linux/ramips/mt76x8/base-files/etc/board.d/02_network
@@ -56,6 +56,7 @@
 	tplink,tl-mr3420-v5|\
 	tplink,tl-wr840n-v4|\
 	tplink,tl-wr840n-v5|\
+	tplink,tl-wr840n-v6.2|\
 	tplink,tl-wr841n-v13|\
 	tplink,tl-wr841n-v14|\
 	tplink,tl-wr842n-v5|\
@@ -323,6 +324,7 @@
 	tplink,tl-mr3420-v5|\
 	tplink,tl-wr840n-v4|\
 	tplink,tl-wr840n-v5|\
+	tplink,tl-wr840n-v6.2|\
 	tplink,tl-wr841n-v13|\
 	tplink,tl-wr841n-v14|\
 	tplink,tl-wr842n-v5|\
